azkaban-common/src/test/java/azkaban/executor/ExecutorManagerTest.java

```diff
@@ -349,4 +386,18 @@ private void testSetUpForRunningFlows()
     this.activeFlows.put(this.flow2.getExecutionId(), new Pair<>(ref2, this.flow2));
     when(this.loader.fetchActiveFlows()).thenReturn(this.activeFlows);
   }
+
+  private ExecutableFlow waitFlowFinished(final ExecutableFlow flow1) throws Exception {
+    ExecutableFlow fetchedFlow;
+    for (int i = 0; i < 1000; i++) {
+      fetchedFlow = this.loader.fetchExecutableFlow(flow1.getExecutionId());
+      if (fetchedFlow != null && Status.isStatusFinished(fetchedFlow.getStatus())) {
+        return fetchedFlow;
+      }
+      Thread.sleep(10L);
```

Comment:
Actually now that this doesn't use any `notify` mechanism, can be replaced with the use of `Awaitility`. So blocked by https://github.com/azkaban/azkaban/pull/1442 to add that dependency.